File: bundle/BarcodeReaderCLI/examples/brcli-example.js
// node brcli-example.js           ->  use BarcodeReaderCLI command line 
// node brcli-example.js config    ->  use BarcodeReaderCLI configuration file  

var cp = require('child_process');
process.chdir(__dirname);

var exe = '../bin/BarcodeReaderCLI';
var args = [];
args.push('-type=code128');
args.push('https://wabr.inliteresearch.com/SampleImages/1d.pdf');

var configFile = "./brcli-example.config";
var argsConfig = []; 
argsConfig.push('@' + configFile);
 
const proc = cp.spawn(exe, process.argv.length > 2 ? argsConfig : args);
proc.stdout.on('data', (data) => {
    console.log(data.toString());
});
proc.stderr.on('data', (data) => {
    console.error(data.toString());
});

File: bundle/BarcodeReaderCLI/examples/brcli-example.py
#  python brcli-example.py           ->  use BarcodeReaderCLI command line 
#  python brcli-example.py config    ->  use BarcodeReaderCLI configuration file  
#  use python 3.5 or newer
#     python3 brcli-example.py           ->  use BarcodeReaderCLI command line 
#     python3 brcli-example.py config    ->  use BarcodeReaderCLI configuration file  

import subprocess
import os
import sys

dir = os.path.dirname(os.path.realpath(__file__))
os.chdir(dir)

exe = '../bin/BarcodeReaderCLI'

args = []
args.append(exe)
args.append('-type=code128')
args.append('https://wabr.inliteresearch.com/SampleImages/1d.pdf')

configFile = './brcli-example.config'
argsConfig = []
argsConfig.append(exe)
argsConfig.append('@' + configFile)

if len(sys.argv) > 1 : args = argsConfig
cp = subprocess.run(args, universal_newlines=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)

output=cp.stdout
error=cp.stderr

if output != "": print("STDOUT:\n" + output)
if error != "":  print("STDERR:\n" + error)
# print("RETURN CODE:"  + str(print(cp.returncode)))